### PR TITLE
node integer delta usage

### PR DESCRIPTION
**Pull request by aie0**
## Types of Changes
Please select the branch type you are merging and fill in the relevant template.

- [ ] Hotfix
- [ ] Release
- [x] Fix or Feature

### Types of Changes

- [x] Tech Debt (Code improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Dependency upgrade (A change in substrate or any 3rd party crate version)

### Migrations and Hooks

- [ ] This change requires a runtime migration.
- [ ] Modifies `on_initialize`
- [ ] Modifies `on_finalize`

### Checklist for Fix or Feature

- [x] Change has been tested locally.
- [x] Change adds / updates tests if applicable.
- [ ] Changelog doc updated.
- [x] `spec_version` has been incremented.
- [ ] `network-relayer`'s [events](https://github.com/Cerebellum-Network/network-relayer/blob/dev-cere/shared/substrate/events.go) have been updated according to the blockchain events if applicable.
- [ ] All CI checks have been passed successfully

## Checklist for Hotfix

- [ ] Change has been deployed to Testnet.
- [ ] Change has been tested in Testnet.
- [ ] Changelog has been updated.
- [ ] Crate version has been updated.
- [ ] `spec_version` has been incremented.
- [ ] Transaction version has been updated if required.
- [ ] Pull Request to `dev` has been created.
- [ ] Pull Request to `staging` has been created.
- [ ] `network-relayer`'s [events](https://github.com/Cerebellum-Network/network-relayer/blob/dev-cere/shared/substrate/events.go) have been updated according to the blockchain events if applicable.
- [ ] All CI checks have been passed successfully

## Checklist for Release

- [ ] Change has been deployed to Devnet.
- [ ] Change has been tested in Devnet.
- [ ] Change has been deployed to Qanet.
- [ ] Change has been tested in Qanet.
- [ ] Change has been deployed to Testnet.
- [ ] Change has been tested in Testnet.
- [ ] Changelog has been updated.
- [ ] Crate version has been updated.
- [ ] Spec version has been updated.
- [ ] Transaction version has been updated if required.
- [ ] All CI checks have been passed successfully

### CODE REVIEW

File: runtime/cere-dev/src/lib.rs [hunk 1]
```diff
@@ -149,7 +149,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	// and set impl_version to 0. If only runtime
 	// implementation changes and behavior does not, then leave spec_version as
 	// is and increment impl_version.
-	spec_version: 54112,
+	spec_version: 54113,
 	impl_version: 0,
 	apis: RUNTIME_API_VERSIONS,
 	transaction_version: 19,
```
Comment (yahortsaryk) [inline runtime/cere-dev/src/lib.rs:155]: You have changed existing extrinsics. The `transaction_version` should be increased..

File: pallets/ddc-nodes/src/migrations.rs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+		Ok((prev_count as u64).encode())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(prev_state: Vec<u8>) -> Result<(), sp_runtime::DispatchError> {
+		let prev_count: u64 =
+			Decode::decode(&mut &prev_state[..]).expect("pre_upgrade provides a valid state; qed");
+
+		let post_count = StorageNodes::<T>::iter().count() as u64;
```
Comment (yahortsaryk) [inline pallets/ddc-nodes/src/migrations.rs:106]: The purpose of the `post_upgrade` is to check that your migration applied the expected changes. You need to fetch the nodes here and check that every node has `total_usage == None`

**Review by yahortsaryk — CHANGES_REQUESTED** [review]

File: runtime/cere-dev/src/lib.rs (hunk 1 above)
Comment (aie0) [reply runtime/cere-dev/src/lib.rs:155]: extrensics were not changed as part of this release and will change in the next one

File: runtime/cere/src/lib.rs
Comment (ayushmishra2005) [inline runtime/cere/src/lib.rs:1]: Please also add new migration in cere runtime

**Review by ayushmishra2005 — CHANGES_REQUESTED** [review]

File: runtime/cere/src/lib.rs [hunk 3]
```diff
@@ -913,6 +913,7 @@ impl pallet_contracts::Config for Runtime {
 		pallet_contracts::migration::v13::Migration<Runtime>,
 		pallet_contracts::migration::v14::Migration<Runtime, Balances>,
 		pallet_contracts::migration::v15::Migration<Runtime>,
+		pallet_ddc_nodes::migrations::MigrateToV1<Runtime>,
```
Comment (ayushmishra2005) [inline runtime/cere/src/lib.rs:916]: This is wrong place. It should be added here https://github.com/Cerebellum-Network/blockchain-node/blob/32dbf724b9a0b182fcb2a373e35644ff355c3dc6/runtime/cere/src/lib.rs#L1418

**Review by ayushmishra2005 — CHANGES_REQUESTED** [review]

**Review by ayushmishra2005 — APPROVED** [review]

File: runtime/cere-dev/src/lib.rs [hunk 4]
```diff
@@ -149,7 +149,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	// and set impl_version to 0. If only runtime
 	// implementation changes and behavior does not, then leave spec_version as
 	// is and increment impl_version.
-	spec_version: 54112,
+	spec_version: 54113,
```
Comment (yahortsaryk) [inline runtime/cere-dev/src/lib.rs:152]: You need to increase `spec_version` in both `cere` and `cere_dev` runtimes as you affected both of them by making changes to the pallet

**Review by yahortsaryk — CHANGES_REQUESTED** [review]

**Review by yahortsaryk — APPROVED** [review]

**Pull request merged**